lib/testobject.js

```diff
@@ -0,0 +1,162 @@
+import zip from './s3';
+import { exec } from 'teen_process';
+import { fs } from 'appium-support';
+import log from '../lib/logger';
+import _ from 'lodash';
+
+const TestObject = {};
+
+TestObject.DEFAULT_DEVICE = 'HTC_One_M8_real';
+TestObject.DEFAULT_IOS_DEVICE = 'iPad_2_16GB_real';
+
+/**
+ * Takes desired capabilities and extends it with testobject capabilities
+ */
+TestObject.getTestObjectCaps = async function (caps = {}) {
+  // Upload the app to TestObject
+  let testobject_app_id;
+  try {
+    testobject_app_id = await TestObject.uploadTestObjectApp(caps.app);
+  } catch (e) {
+    throw new Error(`Could not upload ${caps.app} to TestObject: ${e.message}`);
+  }
+
+  // Determine the TestObject_device
+  let testobject_device;
+  if (process.env.TESTOBJECT_DEVICE) {
+    testobject_device = process.env.TESTOBJECT_DEVICE;
+  } else if (caps.platformName && caps.platformName.toLowerCase() === 'ios') {
+    testobject_device = TestObject.DEFAULT_IOS_DEVICE;
+  } else {
+    testobject_device = TestObject.DEFAULT_DEVICE;
+  }
+
+  // Extend the provided caps with special testobject caps 
+  // (see https://app.staging.testobject.org/#/danielgraham/contact-manager/appium/basic/instructions)
+  return  Object.assign({
+    testobject_app_id,
```

Comment:
yeah, rather than use snake_case throughout, let's assign our local variables to camelCase and then do 

```js
{
  testobject_app_id: testobjectAppId,
  ...
}
```